checker/src/org/checkerframework/checker/index/upperbound/UpperBoundChecker.java

```diff
@@ -17,6 +18,23 @@
 @SuppressWarningsKeys({"index", "upperbound"})
 public class UpperBoundChecker extends BaseTypeChecker {
 
+    private HashSet<String> collectionBaseClasses;
+
+    public UpperBoundChecker() {
+        collectionBaseClasses = new HashSet<>(3);
+        collectionBaseClasses.add("java.util.List");
+        collectionBaseClasses.add("java.util.AbstractList");
+        collectionBaseClasses.add("java.lang.CharSequence");
+    }
+
+    @Override
+    public boolean shouldSkipUses(String typeName) {
```

Comment:
Wouldn't it be easier and clearer to simply use the top annotation for the problematic methods in these three classes?
I'm not quite sure what problem this PR is trying to solve.

Reply:
We want to be able to write these annotations when overriding these methods (to typecheck the body). If we just have top annotations in the base class, we get `override.param.invalid` warnings when writing a more specific annotation in the overriding method.

This PR helps reduce the number of `@SuppressWarnings` that need to be written in Guava when checked by the Index Checker:
- in code overriding `AbstractList` methods
- in code using `CharSequence` (makes consistent the way `CharSequence` and `List` are annotated)

Reply:
If this is only useful for Guava, and other projects that implement these collection classes, how about simply providing the corresponding `-AskipUses=xxx` command-line flags to those projects?
Would the suppressed warnings not be useful for projects that are mere users of the collection classes?
Shouldn't there at least be some comment in the manual about what the Index Checker guarantees for these classes?

Reply:
We cannot use command-line arguments,, because we want this to apply only in the Upper Bound checker, not other subcheckers such as the Lower Bound checker (we want the lower-bound annotations checked).

We consider `List` not to be a fixed-length sequence, because it provides methods that change its length (`CharSequence` is now not included in this PR). The index checker is designed to only check fixed length sequences, so we do not want uses of `List` checked, regardless of whether the object is actually fixed-size or not, or whether the code uses the mutating methods.